network-api/networkapi/wagtailpages/pagemodels/publications/publication.py

```diff
@@ -80,3 +84,31 @@ class PublicationPage(FoundationMetadataPageMixin, Page):
         FieldPanel('contents_title'),
         FieldPanel('notes')
     ]
+
+    @property
+    def is_chapter_page(self) -> bool:
+        """Is this a chapter page (child-Publicationpage). Returns a bool."""
+        parent = self.get_parent().specific
+        return parent.__class__.__name__ == self.__class__.__name__
+
+    def get_chapter_number_or_none(self) -> Union[int, None]:
+        """Return the chapter page number or None."""
+        if self.is_chapter_page:
+            chapter_pages = list(self.get_siblings().specific())
+            return chapter_pages.index(self) + 1
+        return None
+
+    def get_child_article_pages(self) -> Union[QuerySet, list]:
```

Comment:
hm, these next two functions are basically the same. I bet we can consolidate them. 

We already have the subpage_types set up to only allow ArticlePage or PublicationPage depending on the context.

Couldn't we: 
* get live children (my favorite kind of children...)
* check content type of the children
* if their content type is ArticlePage, do the thing. If their content type is Publication page, do the other thing. 

Clearly I'm not 100% sure what exactly we are trying to distinguish here for what purpose, so forgive me. 

(this made me think of an edge case...a publication page could have _some_ Article Pages and _some_ Publication pages, which could be bad? Or is fine?)

Reply:
Correct on the edge case, which is why we're filtering pages with `.type(ArticlePage)` based on if the page is acting as a Chapter Page or Publication Page. This is also a reason to move nested publication pages to their own model type, such as a ChapterIndexPage.